poc/extensions/src/extension_core.rs

```diff
@@ -0,0 +1,28 @@
+use super::{Extension, ExtensionTypeId};
+pub trait ExtensionCore: Extension {
+    // ExtensionId should be generated by the macro
+    // It should normalize the order of methods and parameter names
+    const TYPE_ID: ExtensionTypeId = 0u64;
+    // TODO: Actual args and return values are complex types
+    // and we adapt them to polkavm ABI in `impl XcqExecutorContext for HostFunctions`
+    fn some_host_function() -> u32;
+    fn another_host_function() -> u32;
+}
+
+#[derive(Clone)]
```

Comment:
why clone is needed?

Reply:
Oh, I was supposed to use it then I found it's not necessary and then forgot removing this.